server/game/basecard.js

```diff
@@ -295,7 +299,7 @@ class BaseCard {
     }
 
     hasTrait(trait) {
-        return this.traits.contains(trait);
+        return !this.isBlank(true) && this.traits.contains(trait);
```

Comment:
Will this work, or will traits need to be converted to effects? *shudder*

Reply:
Looks like it should work to me...

Reply:
This is technically incorrect. Consider - a trait-blanked character with the Knighted attachment attached. Traits from the character's text box should be gone, but they should still get the Knight trait from the attachment.

However, since the only thing that blanks traits is White Shadows, I'm not sure it really matters.